### PR TITLE
chore(IT Wallet): [SIW-547] Update `io-react-native-wallet` to `v0.7.0`

### PR DESCRIPTION
**Pull request by balanza**
## Short description
Upgrade package `@pagopa/io-react-native-wallet` to latest version. Change usage accordingly.

See also: https://github.com/pagopa/io-react-native-wallet/pull/43

### CODE REVIEW

**Comment by hevelius** [comment]
Tips: the right way to write a PR title is use this format [SIW-jiraTicketNumber] a short description . In this way Danger add automatically the label itwallet and related type (feat, chore etc...)

File: ts/features/it-wallet/store/actions/itwRpActions.ts [hunk 1]
```diff
@@ -12,7 +12,10 @@ export const itwRpInitialization = createAsyncAction(
   "ITW_RP_INITIALIZATION_FAILURE"
 )<
   { authReqUrl: string; clientId: string },
-  { requestObject: RequestObjectConf; entity: RpEntityConfiguration },
+  {
+    requestObject: RequestObjectConf;
+    entity: Awaited<ReturnType<typeof getRelyingPartyEntityConfiguration>>;
```
Comment (hevelius) [inline ts/features/it-wallet/store/actions/itwRpActions.ts:17]: Why not use directly the type `RelyingPartyEntityConfiguration` instead? https://github.com/pagopa/io-react-native-wallet/blob/72a467154f702e49e57376437b6986f802d89f74/src/trust/types.ts#L152

File: ts/features/it-wallet/store/reducers/itwRpInitializationReducer.ts [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 import { Action } from "../../../../store/actions/types";
 import { ItWalletError } from "../../utils/errors/itwErrors";
 import { GlobalState } from "../../../../store/reducers/types";
 import { itwRpInitialization } from "../actions/itwRpActions";
 
 export type ItwRpInitializationType = {
   requestObject: O.Option<RequestObjectConf>;
-  entity: O.Option<RpEntityConfiguration>;
+  entity: O.Option<
+    Awaited<ReturnType<typeof getRelyingPartyEntityConfiguration>>
```
Comment (hevelius) [inline ts/features/it-wallet/store/reducers/itwRpInitializationReducer.ts:14]: ditto

File: ts/features/it-wallet/utils/wia.ts [hunk 3]
```diff
@@ -13,13 +13,21 @@ export const ITW_WIA_KEY_TAG = "ITW_WIA_CRYTPO";
...
  * Getter for the Wallet Instance Attestation.
  * It generates a new key pair and uses it to sign the attestation request.
+ *
+ * @param entityConfiguration The entity configuration object for the wallet provider
+ *
  * @returns the WIA attestation
  */
-export const getWia = async () => {
+export const getWia = async (
+  entityConfiguration: Awaited<
```
Comment (hevelius) [inline ts/features/it-wallet/utils/wia.ts:22]: ditto

**Review by hevelius — CHANGES_REQUESTED** [review]

File: ts/features/it-wallet/store/actions/itwRpActions.ts (hunk 1 above)
Comment (balanza) [reply ts/features/it-wallet/store/actions/itwRpActions.ts:17]: I guess it's not exposed anymore

Comment (balanza) [reply ts/features/it-wallet/store/actions/itwRpActions.ts]: solved

**Review by LazyAfternoons — APPROVED** [review]
LGTM

**Review by hevelius — APPROVED** [review]
LGTM. It works as expected.

**Pull request merged**